src/Castle.Windsor/MicroKernel/Arguments.cs

```diff
@@ -19,208 +19,240 @@ namespace Castle.MicroKernel
 	using System.Collections.Generic;
 
 	using Castle.Core;
-	using Castle.MicroKernel.Context;
+	using Castle.Windsor;
 
 	/// <summary>
-	///   Represents collection of arguments used when resolving a component.
+	/// Represents a collection of named and typed arguments used for dependencies resolved via <see cref="IWindsorContainer.Resolve{T}(Castle.MicroKernel.Arguments)"/>
+	/// Please see: https://github.com/castleproject/Windsor/blob/master/docs/arguments.md
 	/// </summary>
 	public class Arguments : IDictionary
 	{
-		protected IDictionary arguments;
+		private static readonly ArgumentsComparer Comparer = new ArgumentsComparer();
 
-		public Arguments(object namedArgumentsAsAnonymousType, params IArgumentsComparer[] customComparers)
-			: this(new ReflectionBasedDictionaryAdapter(namedArgumentsAsAnonymousType), customComparers)
+		private readonly IDictionary dictionary;
+		private bool isReadOnly;
+
+		/// <summary>
+		/// Constructor for creating named/typed dependency arguments for <see cref="IWindsorContainer.Resolve{T}(Castle.MicroKernel.Arguments)"/>
+		/// </summary>
+		public Arguments()
 		{
+			dictionary = new Dictionary<object, object>(Comparer);
 		}
 
-		public Arguments(IDictionary values, params IArgumentsComparer[] customComparers)
-			: this(customComparers)
+		/// <summary>
+		/// Constructor for creating named dependency arguments for <see cref="IWindsorContainer.Resolve{T}(Castle.MicroKernel.Arguments)"/>
+		/// </summary>
+		/// <param name="name"></param>
+		/// <param name="value"></param>
+		public Arguments(string name, object value) : this()
 		{
-			foreach (DictionaryEntry entry in values)
-			{
-				Add(entry.Key, entry.Value);
-			}
+			dictionary[name] = value;
 		}
 
-		public Arguments(object[] typedArguments, params IArgumentsComparer[] customComparers)
-			: this(customComparers)
+		/// <summary>
+		/// Constructor for creating typed dependency arguments for <see cref="IWindsorContainer.Resolve{T}(Castle.MicroKernel.Arguments)"/>
+		/// </summary>
+		/// <param name="type"></param>
+		/// <param name="value"></param>
+		public Arguments(Type type, object value) : this()
 		{
-			foreach (var @object in typedArguments)
-			{
-				if (@object == null)
-				{
-					throw new ArgumentNullException("typedArguments",
-					                                "Given array has null values. Only non-null values can be used as arguments.");
-				}
-
-				Add(@object.GetType(), @object);
-			}
+			dictionary[type] = value;
 		}
 
-		public Arguments(params IArgumentsComparer[] customComparers)
+		/// <summary>
+		/// Constructor for creating named/typed dependency arguments for <see cref="IWindsorContainer.Resolve{T}(Castle.MicroKernel.Arguments)"/>
+		/// </summary>
+		public Arguments(IDictionary values)
+			: this()
```

Comment:
`: this()` is on a new line here but not with the previous constructor which has a similar line length.